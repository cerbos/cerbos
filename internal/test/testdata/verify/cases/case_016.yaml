description: Principal testdata file not named principals.yaml
