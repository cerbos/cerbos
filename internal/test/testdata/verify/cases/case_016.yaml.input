-- testdata/some_principals.yaml --
---
principals:
  john:
    id: john
    policyVersion: '20210210'
    roles:
      - employee
    attr:
      department: marketing
      geography: GB
      team: design
  bev: &bev
    id: bev
    policyVersion: '20210210'
    roles:
      - employee
      - manager
    attr: &bev_attr
      department: marketing
      geography: GB
      managed_geographies: GB
      ip_address: 10.20.1.2
      team: design
  matt:
    << : *bev
    id: matt
    attr:
      << : *bev_attr
      ip_address: 10.10.1.2

-- testdata/resources.yaml --
---
resources:
  john_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr: &attr
      department: marketing
      geography: GB
      id: XX125
      owner: john
      team: design
  pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX125
    attr:
      << : *attr
      status: PENDING_APPROVAL
  stale_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
  stale_pending_leave_request:
    kind: leave_request
    policyVersion: '20210210'
    id: XX225
    attr:
      << : *attr
      modifiedAt: "2022-08-01T15:00:00Z"
      status: PENDING_APPROVAL

-- testdata/auxdata.yaml --
---
auxData:
  myJWT:
    jwt:
      iss: cerbos-test-suite
      aud: [cerbos-jwt-tests]
      customArray: [A, B]

-- udf_test.yaml --
---
name: TestSuite
description: Tests for verifying something
tests:
  - name: John and his leave request
    input:
      principals:
        - bev
        - matt
      resources:
        - pending_leave_request
      actions:
        - delete
        - approve
    expected:
      - principal: bev
        resource: pending_leave_request
        actions:
          delete: EFFECT_ALLOW
          approve: EFFECT_ALLOW
      - principal: matt
        resource: pending_leave_request
        actions:
          delete: EFFECT_DENY
          approve: EFFECT_ALLOW
