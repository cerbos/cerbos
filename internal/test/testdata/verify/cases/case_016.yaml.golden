{
  "suites": [
    {
      "file": "udf_test.yaml",
      "name": "TestSuite",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "Failed to load the test suite: invalid test \"John and his leave request\": principal \"bev\" not found",
      "description": "Tests for verifying something"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}
